PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ['crm.lead']
-    _inherit = ["event.event"]
-    _inherit = ['event.registration']

Total Changes: 6

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/event_crm/models/crm_lead.py
+++ b/addons/event_crm/models/crm_lead.py
@@ -5,7 +5,7 @@ from odoo import fields, models, api
 
 
 class CrmLead(models.Model):
-    _inherit = ['crm.lead']
+    _inherit = 'crm.lead'
 
     event_lead_rule_id = fields.Many2one('event.lead.rule', string="Registration Rule", help="Rule that created this lead")
     event_id = fields.Many2one('event.event', string="Source Event", help="Event triggering the rule that created this lead")

--- a/addons/event_crm/models/event_event.py
+++ b/addons/event_crm/models/event_event.py
@@ -6,7 +6,7 @@ from odoo.exceptions import UserError
 
 
 class EventEvent(models.Model):
-    _inherit = ["event.event"]
+    _inherit = "event.event"
 
     lead_ids = fields.One2many(
         'crm.lead', 'event_id', string="Leads", groups='sales_team.group_sale_salesman',

--- a/addons/event_crm/models/event_registration.py
+++ b/addons/event_crm/models/event_registration.py
@@ -9,7 +9,7 @@ from odoo.addons.phone_validation.tools import phone_validation
 
 
 class EventRegistration(models.Model):
-    _inherit = ['event.registration']
+    _inherit = 'event.registration'
 
     lead_ids = fields.Many2many(
         'crm.lead', string='Leads', copy=False, readonly=True,
